app/src/main/java/ch/sdp/vibester/activity/QrScanningActivity.kt

```diff
@@ -134,15 +142,18 @@ class QrScanningActivity : AppCompatActivity() {
         }
     }
 
-
-    private fun startActivityWExtra(intent: Intent, name: String?, arg: Serializable?) {
-        if (name != null && arg != null) {
-            intent.putExtra(name, arg)
-        }
+    private fun startActivityWExtra(intent: Intent, name: String, arg: Serializable) {
+        intent.putExtra(name, arg)
         startActivity(intent)
```

Comment:
Now that we have an Intent Switching class, maybe you can refactor it.

Reply:
even delete this method and use directly the intent switching where it is called

Reply:
fixed!